redis-plugin/src/main/java/org/entando/entando/plugins/jpredis/aps/system/redis/LettuceCache.java

```diff
@@ -57,15 +59,28 @@ public ValueWrapper get(Object key) {
         Object localCacheValue = this.getFromLocalCache(key);
         return toValueWrapper(localCacheValue);
     }
-    
+
     private Object getFromLocalCache(Object key) {
-        return this.frontendCache.get(super.getName() + "::" + key.toString());
+        Object value = this.frontendCache.get(super.getName() + "::" + key.toString());
+        return getImmutable(value);
     }
     
     @Override
     public synchronized <T> T get(Object key, Callable<T> valueLoader) {
         logger.warn("Calling remote cache for key {}", key);
         return super.get(key, valueLoader);
     }
-    
+
+    private Object getImmutable(Object value) {
```

Comment:
maybe I'm not understanding but this is not immutable this is a copy, we should use ImmutableList/Map from google collections, so they trigger an error ... or maybe i'm saying something stupid?

Reply:
Method [copyOf()](https://docs.oracle.com/javase/10/docs/api/java/util/List.html#copyOf(java.util.Collection)) of `java.util.Collection` returns an immutable copy, as Guava `ImmutableList`.

> `copyOf​(Collection<? extends E> coll)` | Returns an unmodifiable List containing the elements of  the given Collection, in its iteration order.